cli/sawtooth_cli/admin_command/keygen.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import os
+import sys
+
+from sawtooth_cli.exceptions import CliException
+from sawtooth_signing import pbct as signing
+
+
+def add_keygen_parser(subparsers, parent_parser):
+    """Adds subparser command and flags for 'keygen' command.
+
+    Args:
+        subparsers (:obj:`ArguementParser`): The subcommand parsers.
+        parent_parser (:obj:`ArguementParser`): The parent of the subcomman
+            parsers.
+    """
+    parser = subparsers.add_parser('keygen', parents=[parent_parser])
+
+    parser.add_argument(
+        'key_name',
+        help='name of the key to create',
+        nargs='?')
+
+    parser.add_argument(
+        '--force',
+        help="overwrite files if they exist",
+        action='store_true')
+
+    parser.add_argument(
+        '-q',
+        '--quiet',
+        help="print no output",
+        action='store_true')
+
+
+def do_keygen(args):
+    """Executes the key generation operation, given the parsed arguments.
+
+    Args:
+        args (:obj:`Namespace`): The parsed args.
+    """
+    if args.key_name is not None:
+        key_name = args.key_name
+    else:
+        key_name = 'validator'
+
+    if 'SAWTOOTH_HOME' in os.environ:
+        key_dir = os.path.join(os.environ['SAWTOOTH_HOME'], 'keys')
+    else:
+        key_dir = os.path.join(os.sep, 'etc', 'sawtooth', 'keys')
```

Comment:
This should not use os.sep for same reason as above (it isn't valid on Windows).